### PR TITLE
add support for call hierarchy

### PR DESCRIPTION
**Pull request by daisuzu**
Implemented the feature to find incoming/outgoing call hierarchy.

https://microsoft.github.io/language-server-protocol/specifications/specification-3-16/#textDocument_prepareCallHierarchy

### CODE REVIEW

File: autoload/lsp/ui/vim.vim [hunk 1]
```diff
@@ -344,3 +344,90 @@ function! lsp#ui#vim#code_lens() abort
...
+        return
+    endif
+
+    if lsp#client#is_error(a:data['response']) || !has_key(a:data['response'], 'result')
+        call lsp#utils#error('Failed to '. a:type . ' for ' . a:server . ': ' . lsp#client#error_message(a:data['response']))
+        return
+    endif
+
+    for l:item in a:data['response']['result']
+        call timer_start(1, {x->s:call_hierarchy(a:ctx, a:server, l:item)})
```
Comment (prabirshrestha) [inline autoload/lsp/ui/vim.vim:393]: why do we need a timer here?

**Review by prabirshrestha — CHANGES_REQUESTED** [review]
add comments. thanks for a PR.

Any reason we need a timer?

File: autoload/lsp/ui/vim.vim (hunk 1 above)
Comment (daisuzu) [reply autoload/lsp/ui/vim.vim:393]: It worked without a timer.
I referred to "prepareRename", but I didn't think much about why used a timer...

**Comment by prabirshrestha** [comment]
Merged. Thanks.

**Pull request merged**